### PR TITLE
Fix epoch in send message

### LINKED ISSUE #15: [Bug]: Messages not appearing in UI
### Contact Details

_No response_

### Version

v2.9

### System Type

AREDN node

### What happened?

Installed the latest build (non-release) on a test node. There are no messages displayed. The message database on the node is populated with valid data. 

The output of the javascript console follows:

```
GET
http://172.28.100.200:8080/meshchat/
[HTTP/1.1 200 OK 5ms]

GET
http://172.28.100.200:8080/meshchat/skeleton.css
[HTTP/1.1 200 OK 7ms]

GET
http://172.28.100.200:8080/meshchat/style.css
[HTTP/1.1 200 OK 14ms]

GET
http://172.28.100.200:8080/meshchat/jquery-2.2.0.min.js
[HTTP/1.1 200 OK 31ms]

GET
http://172.28.100.200:8080/meshchat/ohsnap.js
[HTTP/1.1 200 OK 35ms]

GET
http://172.28.100.200:8080/meshchat/js.cookie.js
[HTTP/1.1 200 OK 36ms]

GET
http://172.28.100.200:8080/meshchat/shared.js
[HTTP/1.1 200 OK 39ms]

GET
http://172.28.100.200:8080/meshchat/chat.js
[HTTP/1.1 200 OK 4ms]

Expected declaration but found ‘;’.  Skipped to next declaration. [style.css:306:17](http://172.28.100.200:8080/meshchat/style.css)
GET
http://172.28.100.200:8080/meshchat/skeleton.css
[HTTP/1.1 200 OK 4ms]

GET
http://172.28.100.200:8080/meshchat/style.css
[HTTP/1.1 200 OK 5ms]

Unknown pseudo-class or pseudo-element ‘has’.  Ruleset ignored due to bad selector. [meshchat:1:538](http://172.28.100.200:8080/meshchat/)
Expected declaration but found ‘;’.  Skipped to next declaration. [style.css:306:17](http://172.28.100.200:8080/meshchat/style.css)
XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=config
[HTTP/1.1 200 OK 449ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546583&_=1699546583855
[HTTP/1.1 200 OK 463ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583856
[HTTP/1.1 200 OK 460ms]

GET
http://172.28.100.200:8080/meshchat/alert.mp3
[HTTP/1.1 200 OK 4ms]

Use of drawWindow method from CanvasRenderingContext2D is deprecated. Use tabs.captureTab extensions API instead https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/tabs/captureTab [pagedata-script.js:1:12509](moz-extension://4b5ea91b-b84a-6d4b-b000-0463eaa0f382/content/pagedata-script.js)
GET
http://172.28.100.200:8080/favicon.ico
[HTTP/1.1 404 Not Found 1ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages&call_sign=WT0F&id=xxQBI&epoch=1699546584&_=1699546583857
[HTTP/1.1 200 OK 243ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546598&_=1699546583858
[HTTP/1.1 200 OK 291ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583859
[HTTP/1.1 200 OK 287ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546613&_=1699546583860
[HTTP/1.1 200 OK 297ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583861
```

### What browsers are you seeing the problem on?

Firefox

### PR DESCRIPTION
**Pull request by hickey**
This fixes the problem of creating messages with bad or bogus timestamps when `epoch` is not specified in the `send_message` API call. 

Closes #15

### CODE REVIEW

File: meshchat [hunk 1]
```diff
@@ -95,7 +95,7 @@ function send_message()
     local message = query.message:gsub("\n", "\\n"):gsub('"', '\\"'):gsub("\t", " ")
 
     local epoch = os.time()
-    if tonumber(query.epoch) > epoch then
+    if query.epoch and tonumber(query.epoch) > epoch then
```
Comment (gerner) [inline meshchat:98]: Just to be clear, this means that epoch is not required in the API. And if it's missing you'll get the current time on the server.

That makes sense to me, I just want to make sure these kinds of things are written down somewhere :)

**Review by gerner — APPROVED** [review]
I'm not super familiar with this bug, but LGTM

**Comment by n2mh** [comment]
Would it also make sense to also do a simple check to see if an incoming timestamp is in the future? If so, discard it and use the present time? Or, some other action?

File: meshchat (hunk 1 above)
Comment (hickey) [reply meshchat:98]: Checked the API documentation and added a node about an unspecified `epoch` parameter get set to current node time.

**Pull request merged**